lib/AST/DistributedDecl.cpp

```diff
@@ -163,68 +127,96 @@ Type swift::getDistributedActorSystemType(NominalTypeDecl *actor) {
 
 Type swift::getDistributedActorIDType(NominalTypeDecl *actor) {
   auto &C = actor->getASTContext();
-  return C.getAssociatedTypeOfDistributedSystemOfActor(actor, C.Id_ActorID);
+  return getAssociatedTypeOfDistributedSystemOfActor(actor, C.Id_ActorID);
 }
 
-Type swift::getDistributedActorSystemActorIDType(NominalTypeDecl *system) {
-  assert(!system->isDistributedActor());
-  auto &ctx = system->getASTContext();
+static Type getTypeWitnessByName(NominalTypeDecl *type, ProtocolDecl *protocol,
+                                 Identifier member) {
+  if (!protocol)
+    return ErrorType::get(type->getASTContext());
 
-  auto DAS = ctx.getDistributedActorSystemDecl();
-  if (!DAS)
+  auto module = type->getParentModule();
+  Type selfType = type->getSelfInterfaceType();
+  auto conformance = module->lookupConformance(selfType, protocol);
+  if (!conformance || conformance.isInvalid())
     return Type();
+  return conformance.getTypeWitnessByName(selfType, member);
+}
 
-  // Dig out the serialization requirement type.
-  auto module = system->getParentModule();
-  Type selfType = system->getSelfInterfaceType();
-  auto conformance = module->lookupConformance(selfType, DAS);
-  return conformance.getTypeWitnessByName(selfType, ctx.Id_ActorID);
+Type swift::getDistributedActorSerializationType(
+    DeclContext *actorOrExtension) {
+  auto &ctx = actorOrExtension->getASTContext();
+  auto resultTy = getAssociatedTypeOfDistributedSystemOfActor(
+      actorOrExtension,
+      ctx.Id_SerializationRequirement);
+
+  // Protocols are allowed to either not provide a `SerializationRequirement`
+  // at all or provide it in a conformance requirement.
+  if ((!resultTy || resultTy->hasDependentMember()) &&
+      actorOrExtension->getSelfProtocolDecl()) {
+    auto sig = actorOrExtension->getGenericSignatureOfContext();
+
+    auto actorProtocol = ctx.getProtocol(KnownProtocolKind::DistributedActor);
+    if (!actorProtocol)
+      return Type();
+
+    auto serializationTy =
+        actorProtocol->getAssociatedType(ctx.Id_SerializationRequirement)
+            ->getDeclaredInterfaceType();
+
+    auto protocols = sig->getRequiredProtocols(serializationTy);
```

Comment:
This whole thing is just `sig->getExistentialType(serializationTy)`

Reply:
I got spooked by the comment there, will give it a try!

Reply:
Fixed it, thank you!